jupyter_server_nbmodel/handlers.py

```diff
@@ -247,7 +252,12 @@ async def _execute_snippet(
         with ycell.doc.transaction():
             ycell["execution_count"] = reply_content.get("execution_count")
             ycell["execution_state"] = "idle"
-
+            if metadata["record_timing"]:
+                time_info["shell.execute_reply"] = datetime.now(timezone.utc).isoformat()[:-6]
+                ycell["metadata"]["execution"] = time_info
+    if reply_content["status"]!="ok":
+        time_info["execution_state"] = "failed"
+        ycell["metadata"]["execution"] = time_info
```

Comment:
There is no `execution_state` failed. Instead, it should be something like:

```suggestion
            if metadata["record_timing"]:
                end_time = datetime.now(timezone.utc).isoformat()[:-6]
                if reply_content["status"] == "ok":
                    time_info["shell.execute_reply"] = end_time
                else:
                    time_info["execution_failed"] = end_time
                ycell["metadata"]["execution"] = time_info
```

And there should be a test for `execution_failed` time being set too. I think you can use `raise Exception` to trigger failure.